core/digest.go

```diff
@@ -2,54 +2,82 @@ package core
 
 import (
 	"bytes"
-	"context"
 	"encoding/json"
 	"fmt"
 	"io"
+	"os"
 	"reflect"
 	"sort"
-
+	"time"
+
+	"github.com/dagger/dagger/engine"
+	"github.com/dagger/dagger/engine/buildkit"
+	"github.com/dagger/dagger/engine/sources/gitdns"
+	"github.com/dagger/dagger/engine/sources/httpdns"
+	"github.com/dagger/dagger/network"
+	"github.com/koron-go/prefixw"
+	"github.com/moby/buildkit/identity"
 	"github.com/moby/buildkit/solver/pb"
 	"github.com/opencontainers/go-digest"
-	"github.com/pkg/errors"
 )
 
+var debugDigest = false
+var debugDigestLogs = fmt.Sprintf("/tmp/digests-%d.log", time.Now().UnixNano())
```

Comment:
I don't think we should ever enable these logs outside of a specific dev/debugging session. They helped me troubleshoot the tests for stable service hostnames across sessions, but they're super verbose and pretty specialized, and tracing them irl would probably come with a decent perf hit.

I think they're better left as a crutch for writing/troubleshooting failing tests, and hopefully they're never needed in the wild because we have those tests. :)

Reply:
Oh for sure, I guess the only difference would be whether a dev needs to run `docker exec dagger-engine.dev sh -c 'cat /tmp/digests-*.log'` or `docker logs dagger-engine.dev`, which at the end of the day is not important at all, so this is all good as is.

Reply:
I actually started by having these log to `stderr` like everything else, but for some reason that seemed to be dropping logs, which is pretty alarming in hindsight; maybe I should try to repro that again.

Basically I would `grep` for a hostname from the `TestServiceHostnamesAreStable` test failure and it wouldn't be anywhere in the debug logs (which [print the hostname format of every digest they compute](https://github.com/sipsma/dagger/blob/c5fe3e49aaaaa84b5871981ce552485d951779bf/core/digest.go#L65)), even if I recorded the output with `docker logs -f` before running the test.

When I changed it to write to a file instead, everything worked fine.

(I'm 75% sure it wasn't a PEBKAC.)